app/integrations/discord/interactions/resolvers/ping.rb

```diff
@@ -0,0 +1,28 @@
+module Discord
+  module Interactions
+    module Resolvers
+      class Ping < Resolver
+        EXPECTED_GLOBAL_NAME = 'Discord'.freeze
+        PONG_RESPONSE_TYPE = 1
+
+        Response = Struct.new(:type)
+
+        def call
+          @response_status = RESPONSE_STATUS[:UNAUTHORIZED]
+          @body = {}
+
+          return unless valid_signature? && correct_global_name?
+
+          @response_status = RESPONSE_STATUS[:OK]
+          @body = Response.new(PONG_RESPONSE_TYPE)
```

Comment:
I can't entirely agree with this line for these reasons:

1. `Response` is not the right name for a class meant to contain just the response body attributes. The response body, by definition, is just one piece of the response, so if you want to go this way, then I'd rename it to `ResponseBody` for clarity.

2. If we are building the response body right here in the resolver, then what's the idea of using jbuilder? Building the response body is responsibility of the serializer, so if we are using jbuilder then let's just define any kind of attributes that you need to send to the jbuilder file to build the response, in this case it would be just something like
```rb
@type = PONG_RESPONSE_TYPE
```

If you think it's easier to build the response right here I'm fine with it too, but then let's not use jbuilder for this Discord integration and let's return the `resolver.body.to_json` in the controller.

Reply:
Ok, please take a look!